device/core/src/client_diagnostic.ts

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { Message, errors } from 'azure-iot-common';
+
+/**
+ * Data structure to store diagnostic property data
+ */
+export class DiagnosticPropertyData {
+  // Key of creation time in diagnostic context.
+  static DIAGNOSTIC_CONTEXT_CREATION_TIME_UTC_PROPERTY: string = 'creationtimeutc';
+  // Equal symbol in diagnostic context.
+  static DIAGNOSTIC_CONTEXT_SYMBOL_EQUAL: string = '=';
+  // And symbol in diagnostic context.
+  static DIAGNOSTIC_CONTEXT_SYMBOL_AND: string = '&';
+
+  private diagnosticId: string;
+  private diagnosticCreationTimeUtc: string;
+
+  /**
+   * @constructor
+   */
+  constructor(diagnosticId: string, diagnosticCreationTimeUtc: string) {
+    /* Codes_SRS_DIAGNOSTICPROPERTYDATA_01_001: [If the diagnosticId or diagnosticCreationTimeUtc is null, the constructor shall throw an IllegalArgumentException.] */
+    if (!diagnosticId || !diagnosticCreationTimeUtc) {
+      throw new errors.ArgumentError('The diagnosticId or diagnosticCreationTimeUtc cannot be null or empty.');
+    }
+
+    /* Codes_SRS_DIAGNOSTICPROPERTYDATA_01_002: [The constructor shall save the message body.] */
+    this.diagnosticId = diagnosticId;
+    this.diagnosticCreationTimeUtc = diagnosticCreationTimeUtc;
+  }
+
+  /**
+   * @method            module:azure-iot-device.DiagnosticPropertyData.getDiagnosticId
+   * @description       get the diagnostic id.
+   *
+   * @returns {string}
+   */
+  getDiagnosticId(): string {
+    return this.diagnosticId;
+  }
+
+  /**
+   * @method            module:azure-iot-device.DiagnosticPropertyData.setDiagnosticId
+   * @description       set the diagnostic id.
+   * @param {string}    diagnosticId   The value of diagnostic id.
+   * @throws {ArgumentError}    If value is invalid
+   */
+  setDiagnosticId(diagnosticId: string): void {
+    /* Codes_SRS_DIAGNOSTICPROPERTYDATA_01_003: [A valid diagnosticId shall not be null or empty.] */
+    if (!diagnosticId) {
+      throw new errors.ArgumentError('The diagnosticId cannot be null or empty.');
+    }
+    this.diagnosticId = diagnosticId;
+  }
```

Comment:
unnecessary (make the property public and edit it directly - it's an internal API)

Reply:
removed null check